### PR TITLE
auto3dseg - dints - fix list input for dataset keys

### PR DESCRIPTION
**Pull request by dongyang0122**
Addressed issues arising when running benchmark experiments.

1. Allowed DiNTS algorithm to take list input;
2. Allowed SwinUNETR algorithm to take list input;
3. Updated DiNTS inference scripts for multi-GPU setting;
4. Allowed DiNTS algorithm to take optional input "sigmoid" in the `input.yaml`;
5. Allowed SwinUNETR algorithm to take optional input "sigmoid" in the `input.yaml`.

### CODE REVIEW

**Comment by wyli** [comment]
Integration tests work fine, could you please also review these changes @mingxin-zheng?

**Comment by mingxin-zheng** [comment]
I am now hestitating to include more changes into 1.2. Is this a must-have fix @dongyang0122 ?

**Comment by dongyang0122** [comment]
We have gathered error data through benchmarking a range of diverse datasets. In my opinion, it is essential to integrate these modifications to enhance the reliability of Auto3DSeg.

File: auto3dseg/algorithm_templates/dints/scripts/infer.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 from monai import transforms
 from monai.apps.auto3dseg.auto_runner import logger
 from monai.apps.utils import DEFAULT_FMT
+from monai.auto3dseg.utils import datafold_read
```
Comment (heyufan1995) [inline auto3dseg/algorithm_templates/dints/scripts/infer.py:23]: need to add 
```
import torch.distributed as dist
```

Comment (dongyang0122) [reply auto3dseg/algorithm_templates/dints/scripts/infer.py:24]: Thank you for identifying this one. The necessary correction has been made and the line has now been added.
